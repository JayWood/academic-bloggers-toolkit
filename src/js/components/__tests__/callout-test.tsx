import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Callout from '../callout';

describe('<Callout />', () => {
    it('should match snapshots', () => {
        let component = renderer.create(<Callout children="Hello world" />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(<Callout title="Hello!" children="Hello world" />);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(<Callout isVisible={false} children="Hello world" />);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(<Callout isVisible={false} children="Hello world" />);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(<Callout intent="warning" children="Hello world" />);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        const dismiss = () => void 0;
        component = renderer.create(<Callout dismiss={dismiss}  children="Hello world" />);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
