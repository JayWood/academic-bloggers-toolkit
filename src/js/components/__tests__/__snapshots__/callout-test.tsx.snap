// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Callout /> should match snapshots 1`] = `
<div
  aria-label="Error Hello world"
  className="callout danger"
  role="alert"
>
  <div
    className="callout__heading"
  >
    <h5>
      Error
    </h5>
  </div>
  Hello world
  <style
    jsx={true}
  >
    
                        .callout {
                            border-radius: 3px;
                            padding: 10px 12px 9px;
                            line-height: 1.5;
                            position: relative;
                            font-weight: 500;
                            font-size: 14px;
                            margin: 0 10px 10px;
                        }
                        .callout__heading {
                            display: flex;
                            justify-content: space-between;
                        }
                        h5 {
                            margin: 0 0 5px 0;
                            line-height: 20px;
                            font-weight: 600;
                            padding: 0;
                            font-size: 17px;
                        }
                        button {
                            font-size: 20px;
                            cursor: pointer;
                            border: 0;
                            outline: 0;
                            padding: 0;
                            background: transparent;
                            color: #555d66;
                            position: relative;
                            top: -5px;
                            right: -5px;
                        }
                        button:hover {
                            color: black;
                        }
                        .danger {
                            background-color: rgba(219, 55, 55, 0.15);
                        }
                        .danger h5 {
                            color: #c23030;
                        }
                        .warning {
                            background-color: rgba(217, 130, 43, 0.15);
                        }
                        .warning h5 {
                            color: #bf7326;
                        }
                    
  </style>
</div>
`;

exports[`<Callout /> should match snapshots 2`] = `
<div
  aria-label="Hello! Hello world"
  className="callout danger"
  role="alert"
>
  <div
    className="callout__heading"
  >
    <h5>
      Hello!
    </h5>
  </div>
  Hello world
  <style
    jsx={true}
  >
    
                        .callout {
                            border-radius: 3px;
                            padding: 10px 12px 9px;
                            line-height: 1.5;
                            position: relative;
                            font-weight: 500;
                            font-size: 14px;
                            margin: 0 10px 10px;
                        }
                        .callout__heading {
                            display: flex;
                            justify-content: space-between;
                        }
                        h5 {
                            margin: 0 0 5px 0;
                            line-height: 20px;
                            font-weight: 600;
                            padding: 0;
                            font-size: 17px;
                        }
                        button {
                            font-size: 20px;
                            cursor: pointer;
                            border: 0;
                            outline: 0;
                            padding: 0;
                            background: transparent;
                            color: #555d66;
                            position: relative;
                            top: -5px;
                            right: -5px;
                        }
                        button:hover {
                            color: black;
                        }
                        .danger {
                            background-color: rgba(219, 55, 55, 0.15);
                        }
                        .danger h5 {
                            color: #c23030;
                        }
                        .warning {
                            background-color: rgba(217, 130, 43, 0.15);
                        }
                        .warning h5 {
                            color: #bf7326;
                        }
                    
  </style>
</div>
`;

exports[`<Callout /> should match snapshots 3`] = `null`;

exports[`<Callout /> should match snapshots 4`] = `null`;

exports[`<Callout /> should match snapshots 5`] = `
<div
  aria-label="Warning Hello world"
  className="callout warning"
  role="alert"
>
  <div
    className="callout__heading"
  >
    <h5>
      Warning
    </h5>
  </div>
  Hello world
  <style
    jsx={true}
  >
    
                        .callout {
                            border-radius: 3px;
                            padding: 10px 12px 9px;
                            line-height: 1.5;
                            position: relative;
                            font-weight: 500;
                            font-size: 14px;
                            margin: 0 10px 10px;
                        }
                        .callout__heading {
                            display: flex;
                            justify-content: space-between;
                        }
                        h5 {
                            margin: 0 0 5px 0;
                            line-height: 20px;
                            font-weight: 600;
                            padding: 0;
                            font-size: 17px;
                        }
                        button {
                            font-size: 20px;
                            cursor: pointer;
                            border: 0;
                            outline: 0;
                            padding: 0;
                            background: transparent;
                            color: #555d66;
                            position: relative;
                            top: -5px;
                            right: -5px;
                        }
                        button:hover {
                            color: black;
                        }
                        .danger {
                            background-color: rgba(219, 55, 55, 0.15);
                        }
                        .danger h5 {
                            color: #c23030;
                        }
                        .warning {
                            background-color: rgba(217, 130, 43, 0.15);
                        }
                        .warning h5 {
                            color: #bf7326;
                        }
                    
  </style>
</div>
`;

exports[`<Callout /> should match snapshots 6`] = `
<div
  aria-label="Error Hello world"
  className="callout danger"
  role="alert"
>
  <div
    className="callout__heading"
  >
    <h5>
      Error
    </h5>
    <button
      aria-label="dismiss"
      className="dashicons dashicons-no-alt"
      onClick={[Function]}
    />
  </div>
  Hello world
  <style
    jsx={true}
  >
    
                        .callout {
                            border-radius: 3px;
                            padding: 10px 12px 9px;
                            line-height: 1.5;
                            position: relative;
                            font-weight: 500;
                            font-size: 14px;
                            margin: 0 10px 10px;
                        }
                        .callout__heading {
                            display: flex;
                            justify-content: space-between;
                        }
                        h5 {
                            margin: 0 0 5px 0;
                            line-height: 20px;
                            font-weight: 600;
                            padding: 0;
                            font-size: 17px;
                        }
                        button {
                            font-size: 20px;
                            cursor: pointer;
                            border: 0;
                            outline: 0;
                            padding: 0;
                            background: transparent;
                            color: #555d66;
                            position: relative;
                            top: -5px;
                            right: -5px;
                        }
                        button:hover {
                            color: black;
                        }
                        .danger {
                            background-color: rgba(219, 55, 55, 0.15);
                        }
                        .danger h5 {
                            color: #c23030;
                        }
                        .warning {
                            background-color: rgba(217, 130, 43, 0.15);
                        }
                        .warning h5 {
                            color: #bf7326;
                        }
                    
  </style>
</div>
`;
